pkgs/tools/misc/cyclonedx-python/default.nix

```
@@ -1,65 +1,43 @@
-{ lib
-, fetchFromGitHub
-, python3
+{
+  lib,
+  fetchFromGitHub,
+  python3Packages,
+  nix-update-script,
 }:
 
-let
-  py = python3.override {
-    packageOverrides = self: super: {
-
-      # Requires 'cyclonedx-python-lib = ">= 2.0.0, < 4.0.0"'
-      cyclonedx-python-lib = super.cyclonedx-python-lib.overridePythonAttrs (oldAttrs: rec {
-        version = "3.1.5";
-        src = fetchFromGitHub {
-          owner = "CycloneDX";
-          repo = "cyclonedx-python-lib";
-          rev = "refs/tags/v${version}";
-          hash = "sha256-4lA8OdmvQD94jTeDf+Iz7ZyEQ9fZzCxnXQG9Ir8FKhk=";
-        };
-      });
-    };
-  };
-in
-with py.pkgs;
-
-python3.pkgs.buildPythonApplication rec {
+python3Packages.buildPythonApplication rec {
   pname = "cyclonedx-python";
-  version = "3.11.7";
+  version = "4.5.0";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "CycloneDX";
     repo = "cyclonedx-python";
     rev = "refs/tags/v${version}";
-    sha256 = "sha256-jU/0FkQCyph59TnEE+lckJXsU9whfvWp7dkdfzprYw8=";
+    hash = "sha256-+XeMRREDX1+v+qOeYiHh7uhadfueYYOxspLY3q1NL6s=";
   };
 
-  nativeBuildInputs = with py.pkgs; [
-    poetry-core
-  ];
+  nativeBuildInputs = with python3Packages; [ poetry-core ];
 
-  propagatedBuildInputs = with py.pkgs; [
+  propagatedBuildInputs = with python3Packages; [
     chardet
     cyclonedx-python-lib
     packageurl-python
     pip-requirements-parser
-    setuptools
-    toml
+    packaging
+    tomli
   ];
```

See above

```suggestion
  ] ++ cyclonedx-python-lib.optional-dependencies.validation;
```

https://github.com/CycloneDX/cyclonedx-python/blob/main/pyproject.toml#L72C1-L72C80